helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ZkClientMonitor.java

```diff
@@ -81,8 +90,14 @@ public ZkClientMonitor(String monitorType, String monitorKey, String monitorInst
     if (zkEventThread != null) {
       _zkEventThreadMetric = new ZkThreadMetric(zkEventThread);
     }
+    if (mbeanReservoir == null) {
+      _mbeanReservoir =  new SlidingTimeWindowArrayReservoir(DEFAULT_RESET_INTERVAL_MS, TimeUnit.MILLISECONDS);
```

Comment:
@chenboat  Do we have any data/knowledge about the relationship between the reset interval (`DEFAULT_RESET_INTERVAL_MS `) and the heap usage of Reservoir implementations?

Trying to see if we need to make this reset interval more explicitly configurable?

Reply:
Per this doc: https://metrics.dropwizard.io/4.0.0/manual/core.html, 
"10K measurements / sec with reservoir storing time of 1 minute will take 10000 * 60 * 128 / 8 = 9600000 bytes ~ **9 megabytes**"

So with the current default of 1 hour, it will be about 540 M bytes. One of our moderately loaded prod server heapdump shows a single Reservoir which uses ~200M of memory. We can choose either make either Reservoir or Interval time configurable. I chose the former so that the client can even configure an alternative reservoir type.

Reply:
This makes sense. Thanks for the explanation.